sklearn/decomposition/_nmf.py

```diff
@@ -982,6 +983,9 @@ def non_negative_factorization(X, W=None, H=None, n_components=None, *,
     shuffle : bool, default=False
         If true, randomize the order of coordinates in the CD solver.
 
+    check_input : bool, default=True
+        If False, the input array X will not be checked.
```

Comment:
If we do that, we need to properly detail what  kind of input that function expects.